tests/test_license_files.py

```diff
@@ -0,0 +1,52 @@
+"""
+Skeleton tests for `license_files` from `setup.cfg`.  This is a `setuptools`
+extension which is not yet ratified by the Python community.
+
+`license_files` is the subject of a PEP: https://peps.python.org/pep-0639/
+"""
+
+import pytest
+
+
+@pytest.mark.xfail
+def test_license_files_setupcfg_globs(project) -> None:
+    """
+    Test that file names that contain glob characters are added as globs.
+    """
+    license_files = ["LICEN[CS]E*", "COPYING*", "NOTICE*", "AUTHORS*"]
+    license_files_lst = ", ".join(license_files)
+
+    setup_cfg = f"""\
+[metadata]
+name = test-project
+version = 0.0.1
+license_files = {license_files_lst}
+"""
+
+    project.setup_cfg(setup_cfg)
+    project.setup_py()
+    result = project.generate()
+
+    assert result["project"]["license_files"] == {"globs": license_files}
+
+
+@pytest.mark.xfail
+def test_license_files_setupcfg_files(project) -> None:
+    """
+    Test that file names that list explicitly existing files are added as files.
```

Comment:
Are the files actually supposed to exist in this test? I wasn't sure if you meant to add code that creates them.

Reply:
Yeah, we will have to figure out some way to "expose" the content of those files.

One possibility is that we have a stub-able `write_file` method that we simply replace in our unit tests to dump the file's name and content somewhere for comparison purposes.

I think the `long_description` had a similar problem did it not?

Reply:
Wait, I may have misunderstood: is the intent here that the files should already exist when the plugin runs, or that the plugin should create the files? I read the docstring to imply the former.

Reply:
Realistically, both… if the file exists, we shouldn't be overwriting it should we?

Or do we just overwrite it regardless?

Reply:
Actually, just thinking about this… `setuptools` would have _expected_ the files to already exist if they were listed in `setup.cfg`.

We're simply moving the configuration setting from `setup.cfg` to its new place in `pyproject.toml`, so the files _should_ already be there.  Maybe we should be raising a warning if the file is not found at the time of migration?  Otherwise, `setuptools` should suitably squawk.

Thinking about it now, `README` was a little different -- you could specify the README text inline in the `long_description`, and the code we have migrates this to a separate file and links it in `pyproject.toml`.

Reply:
Whoops, sorry I missed these comments!

I think our philosophy has generally been to stay as close as possible to a "dumb" translation of `setup.cfg` fields into `setup.py` fields. We generally rely on setuptools to validate things as much as possible, because like you said, it _is_ going to check field values as needed when it builds the project. With that in mind, I'd say it makes sense to just copy the file names from `setup.cfg` to `pyproject.toml` without checking whether they exist, or definitely without _failing_ if they don't exist. An mild warning wouldn't be out of place, but personally I would be totally fine just skipping it as well.

I can imagine situations where the file might not exist and it'd be fully legitimate - like that web-based converter we were talking about once upon a time, where people might want to only upload `setup.cfg` and/or `setup.py` but not have to add in other random files that don't actually need to be read during the conversion.

Reply:
...as far as what to actually do here: at minimum I'd suggest rephrasing the docstring. If the intent is to contrast with the other test, which uses globs, and point out that this one does not use globs, perhaps something like this:

```suggestion
    Test that concrete file names are added as paths.
```

Or maybe there's even a better term than "concrete", I'm not sure :shrug: but hopefully something along those lines will be clear enough.